pennylane/devices/default_clifford.py

```diff
@@ -210,6 +254,53 @@ def circuit():
         Maintaining and working with this tableau representation instead of the complete state vector
         makes the calculations of increasingly large Clifford circuits more efficient on this device.
 
+    .. details::
+        :title: Probabilities for Basis States
+        :href: clifford-probabilities
+
+        As the ``default.clifford`` device supports executing quantum circuits with a large number of qubits,
+        we restrict the ability to compute the ``analytical`` probabilities for `all` computational basis
+        states at once to maintain computational efficiency when system size scales beyond ``24`` qubits.
```

Comment:
Could you make it a bit more explicit why probabilities are special in that even though we can efficiently simulate the circuit, there are still exponentially many of them so that is why we need to restrict them